### PR TITLE
Authorization.

### PR DESCRIPTION
**Pull request by muneebulrehman**
# Authorization with cancancan
- Only the owner can create and delete posts
- Only the owner can create and delete comments

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,8 +1,9 @@
 class Post < ApplicationRecord
   after_save :update_post_counter
+  after_destroy :decrement_post_counter
   belongs_to :author, class_name: 'User', foreign_key: 'author_id'
```
Comment (BenMukebo) [inline app/models/post.rb:4]: - [OPTIONAL] You can add this line in `Post Controller` as you did with the `Comment Controller` to make sure to use **CanCanCan** for Post authorization

   ```
     class PostsController < ApplicationController
       load_and_authorize_resource
   
       ...
     end
   ```

**Review by BenMukebo — APPROVED** [review]
<h1>Status: Approve  :champagne:  :+1: :confetti_ball: 

### **Hi @muneebulrehman**, 

### **Congratulations! 🎉 Project Approved ✅**

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 


### To Highlight :1st_place_medal: 
- Your Pull Request has a proper title and description. ✔️  
- No Linter errors :heavy_check_mark: 
- You have Pecfectly  added a role based authorization to the app using [CanCanCan](https://github.com/CanCanCommunity/cancancan). keep rocking :rocket: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [ OPTIONAL] This is a group project. Please let's include all the authors in the README file. The Getting Started section and the License section of [this template](https://github.com/microverseinc/readme-template) are missing in your README file. Please let's make sure that all the main headers in the template are present in your README file. This makes the README file professional and descriptive. 
Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class PostsController < ApplicationController
+  # load_and_authorize_resource only: %i[edit update destroy create]
```
Comment (BenMukebo) [inline app/controllers/posts_controller.rb:2]: - [OPTIONAL] You can add this line in `Post Controller` as you did with the `Comment Controller` to make sure to use **CanCanCan** for Post authorization

   ```
     class PostsController < ApplicationController
       load_and_authorize_resource
   
       ...
     end
   ```

Comment (muneebulrehman) [reply app/controllers/posts_controller.rb:2]: @BenMukebo , thanks a lot for the suggestion. I forgot to uncomment it.

**Pull request merged**